platforms/jvm/toolchains-jvm/src/test/groovy/org/gradle/jvm/toolchain/internal/JvmMetadataWithAddedCapabilities.groovy

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.jvm.toolchain.internal
+
+import org.gradle.internal.jvm.inspection.JavaInstallationCapability
+import org.gradle.internal.jvm.inspection.JvmInstallationMetadata
+
+class JvmMetadataWithAddedCapabilities implements JvmInstallationMetadata {
```

Comment:
💅 Missing javadoc on new top-level type.

Reply:
Perhaps we should have an archunit or checkstyle rule for this.

Reply:
Absolutely.